sklearn/utils/estimator_checks.py

```diff
@@ -1721,3 +1720,26 @@ def check_classifiers_regression_target(name, Estimator):
     e = Estimator()
     msg = 'Unknown label type: '
     assert_raises_regex(ValueError, msg, e.fit, X, y)
+
+
+@ignore_warnings(category=DeprecationWarning)
+def check_decision_proba_consistency(name, Estimator):
+    """
+    Check whether an estimator having both decision_function and
+    predict_proba methods has outputs with perfect rank correlation.
+    """
+    rnd = np.random.RandomState(0)
+    X_train = (3*rnd.uniform(size=(10, 4))).astype(int)
+    y = X_train[:, 0]
+    estimator = Estimator()
+
+    set_testing_parameters(estimator)
+
+    if (hasattr(estimator, "decision_function") and
+            hasattr(estimator, "predict_proba")):
+
+        estimator.fit(X_train, y)
```

Comment:
Nothing seems to be entering this case: I've modified it to say `assert False` but nothing is failing from `sklearn/tests/test_common.py`